crates/storage/nippy-jar/src/lib.rs

```diff
@@ -0,0 +1,411 @@
+use serde::{Deserialize, Serialize};
+use std::{
+    fs::File,
+    io::{Read, Seek, SeekFrom, Write},
+    path::{Path, PathBuf},
+    sync::Mutex,
+};
+use thiserror::Error;
+use zstd::bulk::Decompressor;
+
+pub mod compression;
+use compression::{Compression, Compressors};
+
+const NIPPY_JAR_VERSION: usize = 1;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct NippyJar {
+    /// Version
+    version: usize,
+    /// Number of columns
+    columns: usize,
+    /// Compressor if required
+    compressor: Option<Compressors>,
+    #[serde(skip)]
+    /// Data path for file. Index file will be `{path}.idx`
+    path: Option<PathBuf>,
+    /// soon
+    bloom_filter: bool,
+    /// soon
+    phf: bool,
+}
+
+impl NippyJar {
+    /// Creates new [`NippyJar`].
+    pub fn new(columns: usize, bloom_filter: bool, phf: bool, path: &Path) -> Self {
```

Comment:
do we want to have `path` both as an argument in `new` and in `NippyJar::load`? As far as I understand, we will use `NippyJar::load` to load from the snapshot, and it will overwrite the `path` field anyway.
```suggestion
    pub fn new(columns: usize, bloom_filter: bool, phf: bool) -> Self {
```